torchci/lib/benchmark/compilerUtils.ts

```diff
@@ -222,6 +222,84 @@ export function computeGeomean(
   return returnedGeomean;
 }
 
+export function computeExecutionTime(
+  data: CompilerPerformanceData[],
+  passingModels: { [k: string]: any }
+) {
+  const executionTime: { [k: string]: any } = {};
+  const returnedExecutionTime: any[] = [];
+
+  data.forEach((record: CompilerPerformanceData) => {
+    const bucket = record.granularity_bucket;
+    const workflowId = record.workflow_id;
+    const suite = record.suite;
+    const model = record.name;
+    const absLatency = record.abs_latency;
+
+    // Use clear compiler name to avoid confusion about what they do
+    const compiler =
+      COMPILER_NAMES_TO_DISPLAY_NAMES[record.compiler] ?? record.compiler;
+    if (BLOCKLIST_COMPILERS.includes(compiler)) {
+      return;
+    }
+
+    if (!(bucket in executionTime)) {
+      executionTime[bucket] = {};
+    }
+
+    if (!(workflowId in executionTime[bucket])) {
+      executionTime[bucket][workflowId] = {};
+    }
+
+    if (!(suite in executionTime[bucket][workflowId])) {
+      executionTime[bucket][workflowId][suite] = {};
+    }
+
+    if (!(compiler in executionTime[bucket][workflowId][suite])) {
+      executionTime[bucket][workflowId][suite][compiler] = [];
+    }
+
```

Comment:
[Optional]
ah, it seems like executionTime[bucket][workflowId][suite][compiler] just used to calculate results for returnedExecutionTime,
instead of nested key format maybe store it as flat key and value as:

executionTime[`{bucket}_{workflowId}_{suite}_{compiler}`] = {
     bucket: bucket
     workflowId: workId
     compiler: compiler
     .... other fields 
     absLatency: []
}

then you can do objects.foreach only one layer, and calculate everything

Reply:
This makes the code looks better indeed. I can finally get rid of this hateful nested structure